### PR TITLE
Fixed reloader tests to remove flakiness, simplified them.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: pkg/reloader/reloader.go [hunk 1]
```diff
@@ -162,8 +182,9 @@ func (r *Reloader) Watch(ctx context.Context) error {
 
 // apply triggers Prometheus reload if rules or config changed. If cfgOutputFile is set, we also
 // expand env vars into config file before reloading.
-// Reload is retried in retryInterval until ruleInterval.
+// Reload is retried in retryInterval until watchInterval.
 func (r *Reloader) apply(ctx context.Context) error {
+	fmt.Println("a[[ly")
```
Comment (GiedriusS) [inline pkg/reloader/reloader.go]: Is this needed? :smile:

Comment (bwplotka) [reply pkg/reloader/reloader.go]: Just checking you @GiedriusS  =DDDD

Comment (bwplotka) [reply pkg/reloader/reloader.go]: :man_facepalming:

**Comment by adrien-f** [comment]
Much simpler 👌

**Comment by bwplotka** [comment]
@adrien-f approve? (:

**Pull request merged**